pkl-cli/src/test/kotlin/org/pkl/cli/CliTestRunnerTest.kt

```diff
@@ -487,6 +488,47 @@ class CliTestRunnerTest {
       )
   }
 
+  @Test
+  fun `CliTestRunner locale independence test`(@TempDir tempDir: Path) {
+    val originalLocale = Locale.getDefault()
+    Locale.setDefault(Locale.GERMANY)
+
+    val code =
+        """
+        amends "pkl:test"
+
+        facts {
+            ["localeTest"] {
+                1 == 1
+            }
+        }
+        """
+            .trimIndent()
+    val input = tempDir.resolve("test.pkl").writeString(code).toString()
+    val out = StringWriter()
+    val err = StringWriter()
+    val opts = CliBaseOptions(sourceModules = listOf(input.toUri()), settings = URI("pkl:settings"))
+    val testOpts = CliTestOptions()
+    val runner = CliTestRunner(opts, testOpts, consoleWriter = out, errWriter = err)
+    runner.run()
+
+    assertThat(out.toString().stripFileAndLines(tempDir))
+      .isEqualTo(
+        """
+            module test
+              facts
+                ✔ localeTest
+
+            100.0% tests pass [1 passed], 100.0% asserts pass [1 passed]
+
+            """
+          .trimIndent()
+      )
+    assertThat(err.toString()).isEqualTo("")
+
+    Locale.setDefault(originalLocale)
```

Comment:
These assertions will throw errors, which means that `Locale.setDefault(...)` won't get called if any of these tests fail.

Let's put lines 496 through 527 in a `try/finally`, where `Locale.setDefault` is in the finally so we can ensure that this test cleans up the state that it's mucking with

Reply:
Also note that mutating global JVM state is only safe as long as test execution is single-threaded (which is the default in Gradle).

Reply:
> These assertions will throw errors, which means that Locale.setDefault(...) won't get called if any of these tests fail.

In case the test fails, won't the jvm "shutdown" anyways (and report the error)?
So doesn't this even matter to restore the default in that case? 

> Also note that mutating global JVM state is only safe as long as test execution is single-threaded (which is the default in Gradle).

Do we have the default here? 😅 
When I running tests, it looks like Gradle runs the test parallel.
At least for different Gradle modules.
So I guess changing the Locale might also change the behviour on other modules/tests that runs at that time, correct? 🤔

Reply:
By default, each `Test` task starts one JVM and runs tests in a single thread. I'm not aware of Pkl overriding this default. Multiple `Test` tasks may nevertheless run in parallel.